src/main/java/org/mule/extension/sftp/internal/connection/SftpClient.java

```diff
@@ -419,24 +420,40 @@ public void setPreferredAuthenticationMethods(String preferredAuthenticationMeth
     this.preferredAuthenticationMethods = preferredAuthenticationMethods;
   }
 
-  protected RuntimeException exception(String message, Exception cause) {
-    if (cause instanceof SftpException) {
-      int status = ((SftpException) cause).getStatus();
-      if (status == SSH_FX_CONNECTION_LOST || status == SSH_FX_NO_CONNECTION) {
-        return exception(message, new SftpConnectionException("Error occurred while trying to connect to host",
-                                                              new ConnectionException(cause, owner), CONNECTIVITY, owner));
-      }
-    } else if (cause instanceof IOException) {
-      if (!sftp.isOpen()) {
-        return exception(message, new SftpConnectionException("Error occurred while trying to connect to host",
-                                                              new ConnectionException(cause, owner), CONNECTIVITY, owner));
+  public RuntimeException handleException(String message, Exception cause) {
```

Comment:
Could we add some junit tests to test all this recursive behaviour? We need them to assure ourselves that there is no infinite calls to this method